### ISSUE TITLE
Mismatched input & kernel size 

### ISSUE THREAD

**Issue by BenWard-Cherrier** [clear-nus/VT_SNN#11]
Hi! 
I've just installed everything needed to run your code on Windows 10, and was testing the train_cnn3d.py on your slip_classification data.

I had to make a few changes to the code by putting everything in train_cnn3d.py in a main function to avoid runtime errors to do with threading, but otherwise haven't modified anything. 

When I run the code on the downloaded slip_classification_prepocessed I get the following error: 

Exception has occurred: RuntimeError
shape '[8, 1440]' is invalid for input of size 3840
  File "C:\Users\bw14452\OneDrive - University of Bristol\repos\VT_SNN\vtsnn\models\cnn3d.py", line 97, in forward
    out = out.view([batch_size, np.prod([8, 6, 6, 5])])
  File "C:\Users\bw14452\OneDrive - University of Bristol\repos\VT_SNN\vtsnn\train_cnn3d.py", line 103, in _train
    output = net.forward(*inputs)
  File "C:\Users\bw14452\OneDrive - University of Bristol\repos\VT_SNN\vtsnn\train_cnn3d.py", line 198, in main
    _train(device, net, epoch, optimizer, criterion, train_loader, writer)
  File "C:\Users\bw14452\OneDrive - University of Bristol\repos\VT_SNN\vtsnn\train_cnn3d.py", line 205, in <module>
    main()

I'm running the code with the following parameters: 
python vtsnn/train_cnn3d.py \
 --epochs 100 \
 --lr 0.00001 \
 --sample_file 1 \
 --batch_size 8 \
 --data_dir slip_classification_preprocessed \
 --mode vis \
 --task slip

Note that if I try to preprocess the data myself using preprocess.py on the slip_classification_lite folder with the following parameters: 

python vtsnn/preprocess.py \
  --save_path slip_classification_preprocessed_withcode \
  --data_path slip_classification_lite \
  --tact_threshold 1 \
  --vis_threshold 1 \
  --selection full \
  --bin_duration 0.02 \
  --n_sample_per_object 20 \
  --task slip \
  --network_config network_config/slip_detection
"""

I get a different error when running train_cnn3d.py on that dataset: 
Exception has occurred: RuntimeError
Calculated padded input size per channel: (7 x 63 x 50). Kernel size: (10 x 5 x 5). Kernel size can't be greater than actual input size
  File "C:\Users\bw14452\OneDrive - University of Bristol\repos\VT_SNN\vtsnn\models\cnn3d.py", line 91, in forward
    out = self.conv1(x)
  File "C:\Users\bw14452\OneDrive - University of Bristol\repos\VT_SNN\vtsnn\train_cnn3d.py", line 103, in _train
    output = net.forward(*inputs)
  File "C:\Users\bw14452\OneDrive - University of Bristol\repos\VT_SNN\vtsnn\train_cnn3d.py", line 198, in main
    _train(device, net, epoch, optimizer, criterion, train_loader, writer)
  File "C:\Users\bw14452\OneDrive - University of Bristol\repos\VT_SNN\vtsnn\train_cnn3d.py", line 205, in <module>
    main()

Are there any changes to be made to the parameters I'm using or the formatting of the data to get the code to run?
Thanks in advance!

Best, 
Ben

**Comment by BenWard-Cherrier** [comment]
Hi Tasbolat, 
Do you have any insight on the errors I'm getting?
Cheers, 
Ben

**Comment by BenWard-Cherrier** [comment]
@tasbolat1 
I got the code to run by just changing some of the dimensions of the output layer: 
e.g. "self.fc = nn.Linear(np.prod([8, 6, 6, 5]), output_size)" -> "self.fc = nn.Linear(np.prod([8, 2, 6, 5]), output_size)" in the VisCNN3D class when running in --mode vis.

Another small tweak that needed to be made was to convert the labels to torch.long in the loss calculation function. I just added: 
"label = label.long()"
before the
"loss = criterion(output, label)"
in the _train and _test functions of train_cnn3d.py.

Code now seems to be running ok, I'll now check the outputs to make sure it's performing as expected.
